src/genie/libs/parser/iosxe/show_snmp.py

```diff
@@ -80,6 +81,329 @@ def cli(self, output=None):
+                if not snmp_obj.get("snmp_output"):
+                    snmp_obj["snmp_output"] = {}
+                snmp_obj["snmp_output"].update({ "packet_count": int(match.group("output_count"))})
+                continue
+            # 0 Too big errors (Maximum packet size 1500)
+            elif p_output_big.match(line):
+                match = p_output_big.match(line)
+                snmp_obj["snmp_output"].update({"too_big_errors": int(match.group("output_big"))})
+                continue
+            # 1 No such name errors
+            elif p_output_name.match(line):
+                match = p_output_name.match(line)
+                snmp_obj["snmp_output"].update({"no_such_name_errors": int(match.group("output_name"))})
+                continue
+            # 0 Bad values errors
+            elif p_output_bad.match(line):
+                match = p_output_bad.match(line)
+                snmp_obj["snmp_output"].update({"bad_value_errors": int(match.group("output_bad"))})
+                continue
+            # 0 General errors
+            elif p_output_general.match(line):
+                match = p_output_general.match(line)
+                snmp_obj["snmp_output"].update({"general_errors": int(match.group("output_general"))})
+                continue
+            # 322899 Response PDUs
+            elif p_output_response_pdu.match(line):
+                match = p_output_response_pdu.match(line)
+                snmp_obj["snmp_output"].update({"response_pdus": int(match.group("output_response_pdu"))})
+                continue
+            # 527788 Trap PDUs
+            elif p_output_trap_pdu.match(line):
+                match = p_output_trap_pdu.match(line)
+                snmp_obj["snmp_output"].update({"trap_pdus": int(match.group("output_trap"))})
+                continue
+            # SNMP global trap: enabled
+            elif p_global_trap.match(line):
+                snmp_obj["snmp_global_trap"] = "enabled"
+                continue
+            # Packets currently in SNMP process input queue: 0
+            elif p_packet_queue.match(line):
+                match = p_packet_queue.match(line)
+                snmp_obj["snmp_input_queue"] = int(match.group("queue"))
+                continue
+            # SNMP logging: enabled
+            elif p_logging.match(line):
+                match = p_logging.match(line)
+                logging_status = match.group("logging_status")
+                if logging_status == "disabled":
+                    snmp_obj.update({"snmp_logging" : {"status" : "disabled"}})
+                else:
+                    snmp_obj.update({"snmp_logging" : {"status" : "enabled"}})
+                continue
+            # Logging to 13.43.1.12.151, 0/10, 102045 sent, 3065 dropped.
+            elif p_logging_endpoint.match(line):
+                match = p_logging_endpoint.match(line)
+                if not snmp_obj["snmp_logging"].get("endpoints"):
```

Comment:
Also use `setdefault()` here to prevent any potential key errors with `snmp_obj["snmp_logging"]`.

Reply:
Updated